crates/ruma-signatures/src/keys.rs

```diff
@@ -19,21 +24,61 @@ pub trait KeyPair: Sized {
     fn sign(&self, message: &[u8]) -> Signature;
 }
 
+pub const ED25519_OID: ObjectIdentifier = ObjectIdentifier::new("1.3.101.112");
```

Comment:
Do we really need to declare this constant here to create the right `AlgorithmIdentifier`? Should it maybe have an `ed25519` constructor in the `pkcs8` crate (or this constant be moved to that crate)?

Reply:
that crate is agnostic to the actual algorithms (see https://github.com/RustCrypto/utils/pull/424#issuecomment-846200463), it only provides methods to decode the DER-encoded ASN.1 documents, in this case a PKCS#8 document, but furthermore it wont say or do anything with the data contained, so it comes to "us" to do this.

ed25517-dalek should contain that const, but honestly, i'm not thrilled to try to merge it into there for the moment, the maintainer hasn't said anything yet about my current issues...